e2e/testcases/helm_sync_test.go

```diff
@@ -136,6 +137,131 @@ func TestPublicHelm(t *testing.T) {
 	}
 }
 
+// TestHelmWatchConfigMap can run on both Kind and GKE clusters.
+// It tests that helm-sync properly watches ConfigMaps in the RSync namespace and pulls updated values.
+func TestHelmWatchConfigMap(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.Unstructured)
+
+	configMapFilePath := "../testdata/root-sync-helm-cm-update/configmap-1.yaml"
+	nt.T.Logf("Apply the ConfigMap with values to the cluster defined in %s", configMapFilePath)
+	nt.MustKubectl("apply", "-f", configMapFilePath)
+
+	nt.T.Log("Update RootSync to sync from a public Helm Chart with specified release namespace and multiple inline values")
+	rootSyncFilePath := "../testdata/root-sync-helm-cm-update/root-sync.yaml"
+	nt.T.Logf("Apply the RootSync object defined in %s", rootSyncFilePath)
+	nt.MustKubectl("apply", "-f", rootSyncFilePath)
+
+	err := nt.WatchForAllSyncs(nomostest.WithRootSha1Func(helmChartVersion("15.2.35")),
+		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: "wordpress"}))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	var expectedCPURequest string
+	var expectedCPULimit string
+	var expectedMemoryRequest string
+	var expectedMemoryLimit string
+	if nt.IsGKEAutopilot {
+		expectedCPURequest = "250m"
+		expectedCPULimit = expectedCPURequest
+		expectedMemoryRequest = "512Mi"
+		expectedMemoryLimit = expectedMemoryRequest
+	} else {
+		expectedCPURequest = "150m"
+		expectedCPULimit = "1"
+		expectedMemoryRequest = "250Mi"
+		expectedMemoryLimit = "300Mi"
+	}
+	if err := nt.Validate("my-wordpress", "wordpress", &appsv1.Deployment{}, containerImagePullPolicy("Always"),
+		testpredicates.HasCorrectResourceRequestsLimits("wordpress",
+			resource.MustParse(expectedCPURequest),
+			resource.MustParse(expectedCPULimit),
+			resource.MustParse(expectedMemoryRequest),
+			resource.MustParse(expectedMemoryLimit)),
+		testpredicates.HasExactlyImage("wordpress", "bitnami/wordpress", "", "sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_USERNAME", "test-user-1"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_EMAIL", "test-user@example.com"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_1", "val1"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_2", "val2")); err != nil {
+		nt.T.Error(err)
+	}
+	if nt.T.Failed() {
+		nt.T.FailNow()
+	}
+
+	configMapFilePath = "../testdata/root-sync-helm-cm-update/configmap-2.yaml"
+	nt.T.Logf("Update the ConfigMap with values to the cluster defined in %s", configMapFilePath)
+	nt.MustKubectl("apply", "-f", configMapFilePath)
+
+	// Config Sync takes about 4 minutes to pick up the update, rehydrate, and sync the resources to the cluster.
+	// Since we cannot use the RootSync status to detect the sync, we wait 7 minutes as a buffer.
+	time.Sleep(7 * time.Minute)
```

Comment:
Use `WatchObject` (with an increased timeout, if needed) instead of sleep + validate